mindmaps-migration/json/src/main/java/io/mindmaps/migration/json/JsonDataMigrator.java

```diff
@@ -0,0 +1,155 @@
+        Instance instance = isResource ?
+                migrateResource(jsonType, type.asResourceType(), data) : migrateEntity(type.asEntityType());
+
+        switch (jsonType) {
+            case OBJECT:
+                migrateJsonObject(instance, data);
+                break;
+            case ARRAY:
+                migrateJsonArray(instance, data);
+                break;
+            case NULL:
+                break;
+        }
+
+        return instance;
+    }
+
+    private void migrateJsonObject(Instance instance, Json data) {
+        data.asJsonMap().forEach((key, value) -> migrateProperty(instance, key, value));
+    }
+
+    private void migrateJsonArray(Instance instance, Json data) {
+        String propertyName = JsonMapper.arrayItemName(instance.type());
+        data.asJsonList().forEach(item -> migrateProperty(instance, propertyName, item));
+    }
+
+    private void migrateProperty(Instance instance, String propertyName, Json data) {
+        Type propertyType = graph.getType(propertyName);
+
+        RelationType relationType;
+        RoleType roleOwner;
+        RoleType roleOther;
+        if(propertyType.isResourceType()) {
+            relationType = graph.getRelationType(JsonMapper.resourceRelationName(propertyType));
+            roleOwner = graph.getRoleType(JsonMapper.roleOwnerResourceName(propertyType));
+            roleOther = graph.getRoleType(JsonMapper.roleOtherResourceName(propertyType));
+        }
+        else {
+            relationType = graph.getRelationType(JsonMapper.relationName(propertyType));
+            roleOwner = graph.getRoleType(JsonMapper.roleOwnerName(propertyType));
+            roleOther = graph.getRoleType(JsonMapper.roleOtherName(propertyType));
+        }
+
+        Instance propertyInstance = migrateData(propertyType, data);
+
+        Relation relation = graph.putRelation(UUID.randomUUID().toString(), relationType);
+        relation.putRolePlayer(roleOwner, instance);
+        relation.putRolePlayer(roleOther, propertyInstance);
+    }
+
+    private Instance migrateEntity(EntityType type){
+        String id = UUID.randomUUID().toString();
+        return graph.putEntity(id, type);
+    }
+
+    private Instance migrateResource(JsonType jsonType, ResourceType type, Json data){
+
+        switch (jsonType) {
+            case STRING:
+                return migrateJsonString(type, data);
+            case NUMBER:
+            case INTEGER:
+                return migrateJsonNumber(type, data);
+            case BOOLEAN:
+                return migrateJsonBoolean(type, data);
+        }
+
+        return graph.putResource("", type);
+    }
+
+    private Resource migrateJsonString(ResourceType type, Json data) {
+        return graph.putResource(data.asString(), type);
+    }
+
+    private Resource migrateJsonNumber(ResourceType type, Json data) {
+        DataType datatype = type.getDataType();
+        Object value;
```

Comment:
Since this value is always a Long or an Integer you might be able to make this a `Number` rather than an `Object`